python/pyarrow/_flight.pyx

```diff
@@ -211,6 +211,47 @@ cdef class Result:
         return pyarrow_wrap_buffer(self.result.get().body)
 
 
+cdef class BasicAuth:
+    """A container for basic auth."""
+    cdef:
+        unique_ptr[CBasicAuth] basic_auth
+
+    def __init__(self, username=None, password=None):
+        """Create a new basic auth object.
+
+        Parameters
+        ----------
+        username : string
+        password : string
+        """
+        self.basic_auth.reset(new CBasicAuth())
+        if username:
+            self.basic_auth.get().username = tobytes(username)
+        if password:
+            self.basic_auth.get().password = tobytes(password)
+
+    @property
+    def username(self):
+        """Get the username."""
+        return self.basic_auth.get().username
+
+    @property
+    def password(self):
+        """Get the password."""
+        return self.basic_auth.get().password
+
+    @staticmethod
```

Comment:
Maybe for consistency with the equivalent FlightInfo, etc. methods we can call these `serialize` and `deserialize`?

Reply:
agree, fixed